### DISCUSSION TITLE
hi, removeOnComplete: KeepJobs No effect

### DISCUSSION THREAD

**Discussion by Lydanne** [OptimalBits/bull#2320] [Q&A]
https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#keepjobs-options

```ts
const job = await this.transferQueue.add('package', createTransferDto, {
      removeOnFail: true,
      removeOnComplete: {
        age: Math.floor(Date.now() / 1000) + 5,
      },
    });
```

**Comment by manast** [comment]
Can you provide a complete reproducible case? There are many tests that proves this functionality: https://github.com/OptimalBits/bull/blob/develop/test/test_queue.js#L670

**Reply by Pollocks01** [reply]
I did notice when using Bull for the first time today that it removes expired jobs when items are added. If you add an item, it’ll not be purged until you add another item.

**Reply by manast** [reply]
@Pollocks01 no, jobs are only removed when completing or failing and according to the "removeOnComplete" setting.
